azurerm_media_services_account causes plugin to crash

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.9

### AzureRM Provider Version

3.44.1

### Affected Resource(s)/Data Source(s)

azurerm_media_services_account

### Terraform Configuration Files

```hcl
https://livetechgamescom-my.sharepoint.com/:u:/g/personal/will_d_livetechgames_com/Efw4vHYMkDBCvk7QgU4nkaEBsBnE8oBIizm3SkNSFy34QQ?e=tSA0TL
```


### Debug Output/Panic Output

```shell
Terraform will perform the following actions:

  # module.example.azurerm_media_services_account.main will be created
  + resource "azurerm_media_services_account" "main" {
      + id                            = (known after apply)
      + location                      = "uksouth"
      + name                          = "ltgexamplemedia1"
      + public_network_access_enabled = true
      + resource_group_name           = "rg-ltg-example-001"
      + storage_authentication_type   = "System"

      + encryption {
          + current_key_identifier = (known after apply)
          + type                   = "SystemKey"

          + managed_identity {
              + use_system_assigned_identity = true
            }
        }

      + identity {
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = "SystemAssigned"
        }

      + key_delivery_access_control {}

      + storage_account {
          + id         = "/subscriptions/<subscription ID here>/resourceGroups/rg-ltg-example-001/providers/Microsoft.Storage/storageAccounts/saexamplemediaservice"
          + is_primary = true

          + managed_identity {
              + use_system_assigned_identity = true
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────────────────────────

Saved the plan to: tf.plan

To perform exactly these actions, run the following command to apply:
    terraform apply "tf.plan"

🥞 >terraform apply "tf.plan"
module.example.azurerm_media_services_account.main: Creating...
╷
│ Error: Plugin did not respond
│
│   with module.example.azurerm_media_services_account.main,
│   on ../../main.tf line 24, in resource "azurerm_media_services_account" "main":
│   24: resource "azurerm_media_services_account" "main" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin
│ logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v3.44.1_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 127 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/media.expandMediaServicesAccountKeyDelivery({0x1400166b310?, 0x1077a4344?, 0x10902e420?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/media/media_services_account_resource.go:530 +0x424
github.com/hashicorp/terraform-provider-azurerm/internal/services/media.resourceMediaServicesAccountCreateUpdate(0x14002494b00, {0x1084f5180?, 0x140020ee000})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/media/media_services_account_resource.go:228 +0x834
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x10903ce60?, {0x10903ce60?, 0x140009d5e30?}, 0xd?, {0x1084f5180?, 0x140020ee000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:695 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140012c0540, {0x10903ce60, 0x140009d5e30}, 0x14001da8b60, 0x14002494980, {0x1084f5180, 0x140020ee000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400060eed0, {0x10903ce60?, 0x140009d5d10?}, 0x14001f2b360)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xb70
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140009d9ea0, {0x10903ce60?, 0x140009d5230?}, 0x14002277340)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x3b8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x108c14f40?, 0x140009d9ea0}, {0x10903ce60, 0x140009d5230}, 0x140022772d0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140000ba000, {0x10904d300, 0x14001604820}, 0x14001d9f680, 0x1400133b620, 0x10d47c6a0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xb7c
google.golang.org/grpc.(*Server).handleStream(0x140000ba000, {0x10904d300, 0x14001604820}, 0x14001d9f680, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x290

Error: The terraform-provider-azurerm_v3.44.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin\'s maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Not error and deploy a media service account resource

### Actual Behaviour

Plugin crashes

### Steps to Reproduce

cd examples/basic-use
tf init
tf plan -out tf.plan
terraform apply tf.plan

### Important Factoids

N/A

### References

_No response_

--- a/internal/services/media/media_services_account_resource.go
+++ b/internal/services/media/media_services_account_resource.go
@@ -526,10 +526,8 @@ func expandMediaServicesAccountKeyDelivery(input []interface{}) *accounts.KeyDel
 	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
-	keyDelivery, ok := input[0].(map[string]interface{})
-	if !ok {
-		return nil
-	}
+	
+	keyDelivery := input[0].(map[string]interface{})
 	defaultAction := keyDelivery["default_action"].(string)
 
 	var ipAllowList *[]string

--- a/internal/services/media/media_services_account_resource.go
+++ b/internal/services/media/media_services_account_resource.go
@@ -523,10 +523,9 @@ func flattenMediaServicesAccountIdentity(input *accounts.MediaServiceIdentity) (
 }
 
 func expandMediaServicesAccountKeyDelivery(input []interface{}) *accounts.KeyDelivery {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
-
 	keyDelivery, ok := input[0].(map[string]interface{})
 	if !ok {
 		return nil

--- a/internal/services/media/media_services_account_resource.go
+++ b/internal/services/media/media_services_account_resource.go
@@ -526,7 +526,7 @@ func expandMediaServicesAccountKeyDelivery(input []interface{}) *accounts.KeyDel
 	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
-	
+
 	keyDelivery := input[0].(map[string]interface{})
 	defaultAction := keyDelivery["default_action"].(string)
 

--- a/internal/services/media/media_services_account_resource.go
+++ b/internal/services/media/media_services_account_resource.go
@@ -527,7 +527,10 @@ func expandMediaServicesAccountKeyDelivery(input []interface{}) *accounts.KeyDel
 		return nil
 	}
 
-	keyDelivery := input[0].(map[string]interface{})
+	keyDelivery, ok := input[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
 	defaultAction := keyDelivery["default_action"].(string)
 
 	var ipAllowList *[]string

--- a/internal/services/media/media_services_account_resource_test.go
+++ b/internal/services/media/media_services_account_resource_test.go
@@ -340,7 +340,7 @@ resource "azurerm_key_vault_access_policy" "client" {
   key_vault_id       = azurerm_key_vault.test.id
   tenant_id          = data.azurerm_client_config.current.tenant_id
   object_id          = data.azurerm_client_config.current.object_id
-  key_permissions    = ["Get", "Create", "Delete", "List", "Restore", "Recover", "UnwrapKey", "WrapKey", "Purge", "Encrypt", "Decrypt", "Sign", "Verify"]
+  key_permissions    = ["Get", "Create", "Delete", "List", "Restore", "Recover", "UnwrapKey", "WrapKey", "Purge", "Encrypt", "Decrypt", "Sign", "Verify", "GetRotationPolicy"]
   secret_permissions = ["Get"]
 }
 

